src/thor/unidirectional_astar.cc

```diff
@@ -136,57 +149,104 @@ bool TimeDepForward::ExpandForward(GraphReader& graphreader,
     // Decide if we should expand a shortcut or the non-shortcut edge...
 
     // We didn't add any shortcut of the uturn, therefore evaluate the regular uturn instead
-    disable_uturn = ExpandForwardInner(graphreader, pred, nodeinfo, pred_idx, uturn_meta, tile,
-                                       offset_time, destination, best_path) ||
+    disable_uturn = ExpandInner(graphreader, pred, opp_pred_edge, nodeinfo, pred_idx, uturn_meta,
+                                tile, offset_time, destination, best_path) ||
                     disable_uturn;
   }
 
   return disable_uturn;
 }
 
-// Runs in the inner loop of `ExpandForward`, essentially evaluating if
+// Runs in the inner loop of `Expand`, essentially evaluating if
 // the edge described in `meta` should be placed on the stack
 // as well as doing just that.
 //
 // Returns true if any edge _could_ have been expanded after restrictions etc.
-inline bool TimeDepForward::ExpandForwardInner(GraphReader& graphreader,
-                                               const EdgeLabel& pred,
-                                               const NodeInfo* nodeinfo,
-                                               const uint32_t pred_idx,
-                                               const EdgeMetadata& meta,
-                                               const graph_tile_ptr& tile,
-                                               const TimeInfo& time_info,
-                                               const valhalla::Location& destination,
-                                               std::pair<int32_t, float>& best_path) {
+template <const ExpansionType expansion_direction, const bool FORWARD>
+inline bool UnidirectionalAStar<expansion_direction, FORWARD>::ExpandInner(
+    GraphReader& graphreader,
+    const BDEdgeLabel& pred,
+    const baldr::DirectedEdge* opp_pred_edge,
+    const NodeInfo* nodeinfo,
+    const uint32_t pred_idx,
+    const EdgeMetadata& meta,
+    const graph_tile_ptr& tile,
+    const TimeInfo& time_info,
+    const valhalla::Location& destination,
+    std::pair<int32_t, float>& best_path) {
+
+  // Skip shortcut edges for time dependent routes
+  // TODO(danpat): why?
+  if (meta.edge->is_shortcut()) {
+    return false;
+  }
+
+  if (!FORWARD) {
+    // Skip shortcut edges for time dependent routes. Also skip this edge if permanently labeled (best
```

Comment:
This comment seems invalid now. The check below is only checking for access mode.